### PR TITLE
Expansion of the usgscsm_cam_test tool and documentation

### PR DESCRIPTION
**Pull request by oleg-alexandrov**

### CODE REVIEW

File: bin/usgscsm_cam_test.cc
Comment (jessemapel) [inline bin/usgscsm_cam_test.cc:148]: Why return a success/failure code here? If that's the intention, then this should just return a boolean for readability.

File: bin/usgscsm_cam_test.cc
Comment (jessemapel) [inline bin/usgscsm_cam_test.cc:90]: I don't think this is required. For example, you could test every half or tenth of a pixel. The for loops will need to be updated to doubles though.

**Review by jessemapel — COMMENTED** [review]
Looks good, just a few minor comments. It would be good to have a test for the model state dump option. Maybe run it once and dump the state, then run it again and pass in the state file? I'd also like to split these helper functions out into something we can unit test.

File: bin/usgscsm_cam_test.cc
Comment (oleg-alexandrov) [reply bin/usgscsm_cam_test.cc:90]: This block is no longer in the latest version I put up for review, per:  https://github.com/USGS-Astrogeology/usgscsm/pull/379

Comment (jessemapel) [reply bin/usgscsm_cam_test.cc:90]: I'm confused what the test step size has to do with #379. Can you clarify for me?

Comment (oleg-alexandrov) [reply bin/usgscsm_cam_test.cc:90]: I looked at your red text, but you were talking about the green text.

Comment (oleg-alexandrov) [reply bin/usgscsm_cam_test.cc:90]: It is hard to explain to the user what a sample rate of 10.23 is. Everybody understands what a sample rate of 10 is. If a subpixel offset is desired, that one is done with the --subpixel-offset option. So, if your sample rate is 10, and subpixel offset is 0.23, it will test pixels 10.23, 20.23, etc. What you are proposing amounts to testing pixels 10.23, 20.46, etc. I guess the end results are about the same, but with what I propose things may be easier to explain.

File: bin/usgscsm_cam_test.cc
Comment (oleg-alexandrov) [reply bin/usgscsm_cam_test.cc]: Fixed in the latest commit.

File: bin/usgscsm_cam_test.cc
Comment (jessemapel) [reply bin/usgscsm_cam_test.cc:90]: Okay that sounds reasonable and the subpixel-offset makes more context this way.

**Review by jessemapel — APPROVED** [review]
I think is good to go once the small merge conflict is resolved. I may put in a PR later on to unit test stuff, but I agree. Right now, everything is fully tested by the two tests.

**Comment by jessemapel** [comment]
I'm okay with merging this if the windows json stuff continue to be a problem

**Comment by oleg-alexandrov** [comment]
Thanks. I think you can pull it in. The windows error is because of getopt.h is missing. That's what I use now to parse command line options. I will think of what to do about that. Maybe using boost, but that is not in libusgscsm's dependencies, and I would not want to add it. I'll try to implement a simple command line parser or something.

**Comment by jessemapel** [comment]
We may be able to use something like this depending on licensing https://github.com/Chunde/getopt-for-windows

**Review by jessemapel — APPROVED** [review]

**Pull request merged**
